parsing/extend.ml

```diff
@@ -14,6 +14,12 @@ type production_position =
   | BorderProd of Constrexpr.side * Gramlib.Gramext.g_assoc option
   | InternalProd
 
+let production_position_eq pos1 pos2 =
+  match pos1, pos2 with
+  | BorderProd (s1,a1), BorderProd (s2,a2) -> s1 = s2 && Option.equal (=) a1 a2
```

Comment:
Here `=` is fine but in order to be future proof, maybe we'd prefer proper equality functions for side and g_assoc?

Reply:
Done, as well as for other added uses of `(=)`.